pkg/frontend/v2/frontend.go

```diff
@@ -222,9 +226,12 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 	retries := f.cfg.WorkerConcurrency + 1 // To make sure we hit at least two different schedulers.
 
 enqueueAgain:
+	level.Debug(spanLogger).Log("msg", "enqueuing request")
```

Comment:
Note to reviewers: I've deliberately added this request started event and the "request succeeded" event below despite adding the span in `enqueueRequest`, as it's possible there's a delay between a request being added to `f.requestsCh` and it being picked up and enqueued by a worker.